server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BootstrapSchemaHistoryRepository.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.core.migrate;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.BeanPropertySqlParameterSource;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.util.CollectionUtils;
+
+import com.oceanbase.odc.common.util.ResourceUtils;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * schema history persistent, the table schema define in migrate_schema_history_table_template.sql
+ * 
+ * @author yizhou.xw
+ * @version : SchemaHistoryRepository.java, v 0.1 2021-03-26 14:26
+ */
+@Slf4j
+public class BootstrapSchemaHistoryRepository implements SchemaHistoryRepository {
```

Comment:
how about JdbcSchemaHistoryRepository ?

Reply:
which is based on scenario, `BootstrapSchemaHistoryRepository` is used on `migrate_schema_history`, the other impl of `SchemaHistoryRepository` can also be based on jdbc but different table.

Reply:
LGTM